### ISSUE TITLE
QR creates wrong authenticator on iPhone X

### ISSUE THREAD

**Issue by dandamian83** [zpao/qrcode.react#95]
![SampleQR](https://user-images.githubusercontent.com/12101673/68778518-f87cff80-063b-11ea-9617-b70c56afc2de.PNG)
I've attached a sample QR code with the generator text: INWTUSRBM5RUO6JX.
If I scan using Google Authenticator for iOs on iPhone X with iOs 13.1.3, I get the following code at some time:
879 257
If I scan the sam QR using Google Authenticator for Android, on a Huawei Mate 10 Pro, Android 9,  at the same time I get the following code: 
210 354
But if I create the authenticator on iOs using the text provided INWTUSRBM5RUO6JX, the authenticator will provide me with the same code (at that same time obviously) as the one from the Android authenticator, which is 210 354.
Nevertheless, going forward, if I try to validate the 2fa code, the 210 354 is the right one.

Need help with this? I have users using iPhones that cannot enable 2fa.
Thanks!

**Comment by zpao** [comment]
Since other apps are generating correct tokens, and the decoded QR Code looks correct, I'm going to say that I suspect there's an issue with Google Authenticator. In fact, this part of their wiki seems particularly damning: https://github.com/google/google-authenticator/wiki/Key-Uri-Format#algorithm. If they're ignoring the algorithm argument and defaulting to a different one than you're expecting, then maybe that is what's causing the issue. Note though that while this open source version exists, they say it's not the same as what they actually ship (and the Android app is a different codebase) so may be a red herring.

**Comment by dandamian83** [comment]
thanks for answering! 
Ok, I followed your reasoning and I tried Google Auth with another QR, the one generated by Binance.com (attached)
![QRBinance](https://user-images.githubusercontent.com/12101673/68833759-eb542500-06bc-11ea-803e-767176eae004.PNG)
The result was that both; iOs Google Authenticator and Android Google Authenticator (on the same phones and OSes as stated in the first comment) returned the same 2fa: 106 251. I think you can try them by yourself using the images.

**Comment by zpao** [comment]
That Binance qr doesn't specify an algorithm param. And the Google authenticator examples do but use sha1, which is the default anyway afaict. Your qr specified sha512. So I suspect you aren't actually testing the bad flow in any of the other examples.

**Comment by dandamian83** [comment]
Ok, so in regards to what you said 

> if they're ignoring the algorithm argument and defaulting to a different one than you're expecting, then maybe that is what's causing the issue. 

and also 

> Your qr specified sha512. 
how can I specify sha1 as the algorithm, so that Google Auth won't get confused on both iOs and Android? I've looked into the docs and didn't find any algo param. 
thanks

**Comment by zpao** [comment]
You're qr decodes to

> otpauth://totp/Venbot.ro?secret=INWTUSRBM5RUO6JX&algorithm=SHA512

I have no idea how you're generating that link but you have an algorithm specified already in there.

**Comment by dandamian83** [comment]
Ok, I solved it. Don't know why but  I was thinking that qrcode is responsible for specifying the algo... then I saw that it was the framework I was using for generating the otpauthURL. So I've just downgraded the algorithm from sha512 to sha1, the default one, as you said, and now works well on iOs too. So yes, the problem was not in qrcode but in the algo param that Google Auth on iOs does not recognizes. 
Thanks for the help!
